framework/caching/DbCache.php

```diff
@@ -192,22 +192,21 @@ protected function getValues($keys)
      */
     protected function setValue($key, $value, $duration)
     {
-        $result = $this->db->noCache(function (Connection $db) use ($key, $value, $duration) {
-            $command = $db->createCommand()
-                ->update($this->cacheTable, [
+        try {
+            $this->db->noCache(function (Connection $db) use ($key, $value, $duration) {
+                $db->createCommand()->upsert($this->cacheTable, [
+                    'id' => $key,
                     'expire' => $duration > 0 ? $duration + time() : 0,
                     'data' => new PdoValue($value, \PDO::PARAM_LOB),
-                ], ['id' => $key]);
-            return $command->execute();
-        });
+                ])->execute();
+            });
 
-        if ($result) {
             $this->gc();
 
             return true;
+        } catch (\Exception $e) {
```

Comment:
What is the reason for hiding this exception?

Reply:
I implemented abstract method: https://github.com/yiisoft/yii2/blob/master/framework/caching/Cache.php#L400-L410

As you see it must return boolean and not throw any exception.

But this is a good question: should we hide an exception in the case of DB layer failures?

@samdark, @rob006 what do you think about this?

Reply:
We could at least log some errors. Same for `addValue()`.

Reply:
Like here? https://github.com/yiisoft/yii2/blob/d880148fd09f827aefee2f418294a72a38579bfb/framework/caching/FileCache.php#L156